### PR TITLE
Go modules

### PR DESCRIPTION
**Pull request by njuettner**
This PR introduces two changes:

* Go modules
* gometalinter is deprecated, instead we could try out golangci-lint

### CODE REVIEW

**Comment by szuecs** [comment]
👍

**Comment by szuecs** [comment]
thanks

**Comment by njuettner** [comment]
@szuecs I've added staticcheck.
The bot also reviews the PR and makes suggestions for code changes according to what linters has been enabled, see:

image

File: delivery.yaml [hunk 1]
```diff
@@ -1,15 +1,23 @@
...
 - id: build
+  overlay: ci/golang
+  cache:
+    paths:
+    - /go/pkg/mod       # pkg cache for Go modules
+    - ~/.cache/go-build # Go build cache
   type: script
   commands:
-  - desc: Build and push docker image
+  - desc: Build docker image
```
Comment (mikkeloscar) [inline delivery.yaml]: We should also run `make test` before building via CDP.

Comment (njuettner) [reply delivery.yaml]: I wasn’t sure about that because we already run make test in travis.

Comment (mikkeloscar) [reply delivery.yaml]: I would say we should have it to ensure we only build and push an image if the tests pass.

**Comment by njuettner** [comment]
👍

**Comment by mikkeloscar** [comment]
:+1:

**Comment by mikkeloscar** [comment]
:+1:

**Comment by njuettner** [comment]
👍

**Pull request merged**